project-base/storefront/components/Pages/CategoryDetail/CategoryDetailProductsWrapper/CategoryDetailProductsWrapper.tsx

```diff
@@ -30,6 +31,25 @@ export const CategoryDetailProductsWrapper: FC<CategoryDetailProductsWrapperProp
     );
     useGtmPaginatedProductListViewEvent(products, gtmProductListName);
 
+    const xxs = useMediaMin('xxs');
+    const xs = useMediaMin('xs');
+    const sm = useMediaMin('sm');
+
+    const imageSize = (() => {
+        switch (true) {
+            case xxs && !xs:
+                return 'extraLarge' as const;
+            case sm:
+                return 'large' as const;
+            default:
+                return 'medium' as const;
+        }
+    })();
```

Comment:
This is hard to read. This might actually be a good place to get the screen width and not `useMediaMin` hook, then refactor this into function `getImageSize(width: number)` and inside compare it using `switch` and sizes in `mediaQueries.ts` (which btw should also be changed to align with the xxs and xs changes).

Reply:
fixed